test/position-managers/PositionManager.modifiyLiquidities.t.sol

```diff
@@ -0,0 +1,294 @@
+        (uint256 liquidity,,) = manager.getPositionInfo(config.poolKey.toId(), positionId);
+
+        assertEq(liquidity, initialLiquidity + newLiquidity);
+    }
+
+    /// @dev hook can decrease liquidity with approval
+    function test_hook_decreaseLiquidity() public {
+        uint256 initialLiquidity = 100e18;
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, initialLiquidity, address(this), ZERO_BYTES);
+
+        // approve the hook for decreasing liquidity
+        lpm.approve(address(hookModifyLiquidities), tokenId);
+
+        // hook decreases liquidity in beforeSwap via hookData
+        uint256 liquidityToDecrease = 10e18;
+        bytes memory calls = getDecreaseEncoded(tokenId, config, liquidityToDecrease, ZERO_BYTES);
+
+        swap(key, true, -1e18, calls);
+
+        bytes32 positionId =
+            Position.calculatePositionKey(address(lpm), config.tickLower, config.tickUpper, bytes32(tokenId));
+        (uint256 liquidity,,) = manager.getPositionInfo(config.poolKey.toId(), positionId);
+
+        assertEq(liquidity, initialLiquidity - liquidityToDecrease);
+    }
+
+    /// @dev hook can collect liquidity with approval
+    function test_hook_collect() public {
+        uint256 initialLiquidity = 100e18;
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, initialLiquidity, address(this), ZERO_BYTES);
+
+        // approve the hook for collecting liquidity
+        lpm.approve(address(hookModifyLiquidities), tokenId);
+
+        // donate to generate revenue
+        uint256 feeRevenue0 = 1e18;
+        uint256 feeRevenue1 = 0.1e18;
+        donateRouter.donate(config.poolKey, feeRevenue0, feeRevenue1, ZERO_BYTES);
+
+        uint256 balance0HookBefore = currency0.balanceOf(address(hookModifyLiquidities));
+        uint256 balance1HookBefore = currency1.balanceOf(address(hookModifyLiquidities));
+
+        // hook collects liquidity in beforeSwap via hookData
+        bytes memory calls = getCollectEncoded(tokenId, config, ZERO_BYTES);
+        swap(key, true, -1e18, calls);
+
+        bytes32 positionId =
+            Position.calculatePositionKey(address(lpm), config.tickLower, config.tickUpper, bytes32(tokenId));
+        (uint256 liquidity,,) = manager.getPositionInfo(config.poolKey.toId(), positionId);
+
+        // liquidity unchanged
+        assertEq(liquidity, initialLiquidity);
+
+        // hook collected the fee revenue
+        assertEq(currency0.balanceOf(address(hookModifyLiquidities)), balance0HookBefore + feeRevenue0 - 1 wei); // imprecision, core is keeping 1 wei
+        assertEq(currency1.balanceOf(address(hookModifyLiquidities)), balance1HookBefore + feeRevenue1 - 1 wei);
+    }
+
+    /// @dev hook can burn liquidity with approval
+    function test_hook_burn() public {
+        // mint some liquidity that is NOT burned in beforeSwap
+        mint(config, 100e18, address(this), ZERO_BYTES);
```

Comment:
Whats the point in this mint that doesnt get burned?

Reply:
because we're burning a position in beforeSwap, if there's only 1 position the swap will fail haha (it was burned JIT)

need some "idle" liquidity thats untouched for the swap to succeed

Reply:
LOL good catch